jetty-util/src/main/java/org/eclipse/jetty/util/log/StdErrLog.java

```diff
@@ -350,28 +341,35 @@ public boolean isDebugEnabled()
     @Override
     public void setDebugEnabled(boolean enabled)
     {
+        Function<StdErrLog, Integer> lvlFunc;
+
         if (enabled)
         {
-            this._level = LEVEL_DEBUG;
-
-            for (Logger log : Log.getLoggers().values())
-            {
-                if (log.getName().startsWith(getName()) && log instanceof StdErrLog)
-                    ((StdErrLog)log).setLevel(LEVEL_DEBUG);
-            }
+            lvlFunc = (logger) -> LEVEL_DEBUG;
         }
         else
         {
-            this._level = this._configuredLevel;
+            lvlFunc = (logger) -> logger.getConfiguredLevel();
+        }
+
+        this.setLevel(lvlFunc.apply(this));
 
-            for (Logger log : Log.getLoggers().values())
+        String name = getName();
+        for (Logger log : Log.getLoggers().values())
+        {
+            if (log.getName().startsWith(name) && log instanceof StdErrLog)
             {
-                if (log.getName().startsWith(getName()) && log instanceof StdErrLog)
-                    ((StdErrLog)log).setLevel(((StdErrLog)log)._configuredLevel);
+                StdErrLog logger = (StdErrLog)log;
+                logger.setLevel(lvlFunc.apply(logger));
```

Comment:
This seems to me more complicated than it needs to.
Was not enough to use a local variable?

```java
StdErrLog logger = (StdErrLog)log;
int level = enabled ? LEVEL_DEBUG : logger.getConfiguredLevel();
logger.setLevel(level);
```

Reply:
Done.